lib/validation.rb

```diff
@@ -304,4 +304,12 @@ def self.validate_from_option_tree(option_tree, option_parents, params)
       end
     end
   end
+
+  def self.validate_postgres_upgrade(postgres_resource:)
```

Comment:
In general, if a method requires a single argument, it should be a positional argument instead of a keyword argument. An exception would be if you definite plans to add other required arguments in the future.

```suggestion
  def self.validate_postgres_upgrade(postgres_resource)
```